qiskit_experiments/visualization/drawers/base_drawer.py

```diff
@@ -0,0 +1,330 @@
+                and no scaling is applied. If nothing is provided, the axis numbers will be
+                displayed in the scientific notation.
+            yval_unit (str): Unit of y values. See ``xval_unit`` for details.
+            figure_title (str): Title of the figure. Defaults to None, i.e. nothing is shown.
+            series_params (Dict[str, Dict[str, Any]]): A dictionary of plot parameters for each series.
+                This is keyed on the name for each series. Sub-dictionary is expected to have following
+                three configurations, "canvas", "color", and "symbol"; "canvas" is the integer index of
+                axis (when multi-canvas plot is set), "color" is the color of the series, and "symbol" is
+                the marker style of the series. Defaults to an empty dictionary.
+            custom_style (PlotStyle): The style definition to use when drawing. This overwrites style
+                parameters in ``default_style`` in :attr:`options`. Defaults to an empty PlotStyle
+                instance (i.e., :code-block:`PlotStyle()`).
+        """
+        return Options(
+            xlabel=None,
+            ylabel=None,
+            xlim=None,
+            ylim=None,
+            xval_unit=None,
+            yval_unit=None,
+            figure_title=None,
+            series_params={},
+            custom_style=PlotStyle(),
+        )
+
+    def set_options(self, **fields):
+        """Set the drawer options.
+        Args:
+            fields: The fields to update the options
+        """
+        self._options.update_options(**fields)
+        self._set_options = self._set_options.union(fields)
+
+    def set_plot_options(self, **fields):
+        """Set the plot options.
+        Args:
+            fields: The fields to update the plot options
+        """
+        self._plot_options.update_options(**fields)
+        self._set_plot_options = self._set_plot_options.union(fields)
+
+    @property
+    def style(self) -> PlotStyle:
+        """The combined plot style for this drawer.
+
+        The returned style instance is a combination of :attr:`options.default_style` and
+        :attr:`plot_options.custom_style`. Style parameters set in ``custom_style`` override those set in
+        ``default_style``. If ``custom_style`` is not an instance of :class:`PlotStyle`, the returned
+        style is equivalent to ``default_style``.
+
+        Returns:
+            PlotStyle: The plot style for this drawer.
+        """
+        if isinstance(self.plot_options.custom_style, PlotStyle):
+            return PlotStyle.merge(self.options.default_style, self.plot_options.custom_style)
+        return self.options.default_style
+
+    @abstractmethod
+    def initialize_canvas(self):
+        """Initialize the drawer canvas."""
+
+    @abstractmethod
+    def format_canvas(self):
+        """Final cleanup for the canvas appearance."""
+
+    @abstractmethod
+    def draw_raw_data(
```

Comment:
I completely agree with renaming the drawer functions to be more generic. I will go ahead and make these changes.

Regarding implementing QVPlotter or not: one of the goals of this framework is that a BasePlotter subclass defines a kind of figure being plotted. This can be a curve-fit/X-Y plot or an IQ level-1 plot. Obviously, this can be changed slightly so that maybe a given subclass is applicable to CurveAnalysis and QV analysis. I can see that there may be additional drawer functions that would allow the module to recreate figure3a, but I'll leave those additional functions for a follow-up PR.

Reply:
Addressed `BaseDrawer` function names in 94158f4.